engine/message/db.go

```diff
@@ -763,65 +779,39 @@ func (db *DB) updateStuckMessages(ctx context.Context, statusFn StatusFunc) erro
 	return nil
 }
 
-const workersPerType = 5
-
 func (db *DB) sendMessagesByType(ctx context.Context, cLock *processinglock.Conn, send SendFunc, q *queue, typ notification.DestType) error {
-	limit := db.c.RateLimit[typ]
-
-	toSend := int(limit.Batch.Seconds() * float64(limit.PerSecond))
-	var failures, processing int
-	sentCount := func() int {
-		return q.SentByType(typ, limit.Batch) - failures
-	}
-
-	sendCh := make(chan *Message, workersPerType)
-	errCh := make(chan error, workersPerType)
-	resCh := make(chan bool, workersPerType)
-	for i := 0; i < workersPerType; i++ {
+	ch := make(chan error)
+	var count int
+	for {
+		msg := q.NextByType(typ)
+		if msg == nil {
+			break
+		}
+		count++
 		go func() {
-			for msg := range sendCh {
-				sent, err := db.sendMessage(ctx, cLock, send, msg)
-				if err != nil {
-					errCh <- err
-				} else {
-					resCh <- sent
-				}
-			}
+			_, err := db.sendMessage(ctx, cLock, send, msg)
+			ch <- err
 		}()
 	}
-	defer close(sendCh)
 
-	var done bool
-	for {
-		if !done && sentCount() < toSend {
-			msg := q.NextByType(typ)
-			if msg == nil {
-				done = true
+	var failed bool
+	for i := 0; i < count; i++ {
+		select {
+		case err := <-ch:
+			if err != nil {
+				log.Log(ctx, fmt.Errorf("send message: %w", err))
+				failed = true
 				continue
 			}
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			case sendCh <- msg:
-				processing++
-			}
-		} else if processing == 0 {
-			break
-		}
-
-		select {
 		case <-ctx.Done():
 			return ctx.Err()
-		case err := <-errCh:
-			return err
-		case sent := <-resCh:
-			if !sent {
-				failures++
-			}
-			processing--
 		}
 	}
 
+	if failed {
+		return errors.New("one or more failures when sending")
```

Comment:
This would just add noise and wouldn't aid in debugging, each full error is already logged independently, this is just to indicate why the top-level method failed.